vendor/engines/ldap_authentication/lib/ldap_authentication/user_entry.rb

```diff
@@ -52,6 +52,16 @@ def to_user
       UserConverter.new(self).to_user
     end
 
+    def self.with_retry(errors = [Net::LDAP::Error], max_attempts: 3)
+      tries = 0
+      begin
+        yield
+      rescue *errors => e
```

Comment:
Since we’re in a method here, we don’t need an explicit `begin` and we can just move the rescue up to the method level, right? Also, given that there is only one caller of `with_retry`, what do you think about just`rescue Net::LDAP::Error` here and simplify the method signature to `(max_attempts=3)` instead of introducing an options hash unnecessarily?

Reply:
We do need the `begin` because otherwise it would retry the whole method and keep resetting `tries` to zero.

I'll clean up the signature.